### PR TITLE
Initial smallbank specification and transaction processor implementation

### PR DESCRIPTION
**Pull request by jsmitchell**

### CODE REVIEW

File: families/smallbank/protos/smallbank.proto [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    enum PayloadType {
+        CREATE_ACCOUNT = 0;
+        DEPOSIT_CHECKING = 1;
+        WRITE_CHECK = 2;
+        TRANSACT_SAVINGS = 3;
+        SEND_PAYMENT = 4;
+        AMALGAMATE = 5;
+    }
+
+    PayloadType payload_type = 1;
```
Comment (peterschwarz) [inline families/smallbank/protos/smallbank.proto]: Remind me why we don't use `oneof` for these?  I know the argument has been made, but I haven't internalized it yet.

Comment (jsmitchell) [inline families/smallbank/protos/smallbank.proto]: I'm using the union pattern, per comments in the design doc.

**Comment by vaporos** [comment]
Are we anticipating multiple smallbank implementations beyond Go?

**Comment by jsmitchell** [comment]
Who knows. Maybe rust?

**Pull request merged**